Compilation errors with DTO, OAuth and ManyToMany with User

##### **Overview of the issue**

Compilation errors below:

```
[ERROR] /Users/theo/Documents/perso/jhipster-many/src/test/java/com/mycompany/myapp/service/mapper/CountryMapperTest.java:[20,45] incompatible types: java.lang.String cannot be converted to java.lang.Long
[ERROR] /Users/theo/Documents/perso/jhipster-many/src/test/java/com/mycompany/myapp/service/dto/CountryDTOTest.java:[14,53] incompatible types: java.lang.String cannot be converted to java.lang.Long
[ERROR] /Users/theo/Documents/perso/jhipster-many/src/test/java/com/mycompany/myapp/service/dto/CountryDTOTest.java:[19,53] incompatible types: java.lang.String cannot be converted to java.lang.Long
```

##### **Motivation for or Use Case**

Project should compile.

##### **Reproduce the error**

Generate the app using the JDL below:

<details>
<summary>JDL file</summary>
<pre>
application {
  config {
    authenticationType oauth2
  }
  entities *
}

entity Country {
    name String
}

relationship ManyToMany {
    Country{User(login)} to User{Country(name)}
}

dto Country with mapstruct</pre>
</details>

##### **Related issues**

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

##### **JHipster Version(s)**

```
jhipster@0.0.1-SNAPSHOT /Users/theo/Documents/perso/jhipster-many/mono
└── generator-jhipster@6.6.0

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "enableHibernateCache": true,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "jhiPrefix": "jhi",
    "languages": ["en", "fr"],
    "messageBroker": false,
    "nativeLanguage": "en",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "serviceDiscoveryType": false,
    "skipClient": false,
    "skipServer": false,
    "testFrameworks": [],
    "websocket": false,
    "authenticationType": "oauth2",
    "jhipsterVersion": "6.6.0",
    "baseName": "jhipster",
    "buildTool": "maven",
    "skipUserManagement": true,
    "clientPackageManager": "npm",
    "applicationType": "monolith",
    "cacheProvider": "ehcache",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "serverPort": "8080",
    "useSass": true,
    "embeddableLaunchScript": false,
    "creationTimestamp": 1579723742179,
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "blueprints": []
  },
  "entities": ["Country"]
}

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Country {
  name String
}
relationship ManyToMany {
  Country{user(login)} to User
}

dto Country with mapstruct
service Country with serviceClass

</pre>
</details>


##### **Environment and Tools**

openjdk version "12.0.2" 2019-07-16
OpenJDK Runtime Environment AdoptOpenJDK (build 12.0.2+10)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.15.1, JRE 12 Mac OS X amd64-64-Bit Compressed References 20190915_201 (JIT enabled, AOT enabled)
OpenJ9   - 0f66c6431
OMR      - ec782f26
JCL      - 9984ea0dd9 based on jdk-12.0.2+10)

git version 2.25.0

node: v10.18.1

npm: 6.13.4

yarn: 1.21.1

Docker version 19.03.5, build 633a0ea

docker-compose version 1.24.1, build 4667896b

-   [x] Checking this box is mandatory (this is just to show you read everything)

--- a/.github/workflows/angular.yml
+++ b/.github/workflows/angular.yml
@@ -67,7 +67,7 @@ jobs:
                       war: 0
                       protractor: 1
                     - app-type: ngx-mariadb-oauth2-infinispan
-                      entity: sql
+                      entity: sqlfull
                       profile: prod
                       war: 0
                       protractor: 1

--- a/generators/entity-server/templates/src/main/java/package/service/dto/EntityDTO.java.ejs
+++ b/generators/entity-server/templates/src/main/java/package/service/dto/EntityDTO.java.ejs
@@ -120,6 +120,7 @@ public class <%= asDto(entityClass) %> implements Serializable {
         const relationshipFieldName = relationships[idx].relationshipFieldName;
         const relationshipFieldNamePlural = relationships[idx].relationshipFieldNamePlural;
         const relationshipType = relationships[idx].relationshipType;
+        const otherEntityName = relationships[idx].otherEntityName;
         const otherEntityNameCapitalized = relationships[idx].otherEntityNameCapitalized;
         const otherEntityFieldCapitalized = relationships[idx].otherEntityFieldCapitalized;
         const ownerSide = relationships[idx].ownerSide;
@@ -132,7 +133,7 @@ public class <%= asDto(entityClass) %> implements Serializable {
     private Set<<%= asDto(otherEntityNameCapitalized) %>> <%= relationshipFieldNamePlural %> = new HashSet<>();
     <%_ } else if (relationshipType === 'many-to-one' || (relationshipType === 'one-to-one' && ownerSide === true)) { _%>
 
-    private <% if (relationshipFieldName === 'user' && authenticationType === 'oauth2') { %>String<% } else { %><%= primaryKeyType %><% } %> <%= relationshipFieldName %>Id;
+    private <% if (otherEntityName === 'user' && authenticationType === 'oauth2') { %>String<% } else { %><%= primaryKeyType %><% } %> <%= relationshipFieldName %>Id;
     <%_ if (otherEntityFieldCapitalized !== 'Id' && otherEntityFieldCapitalized !== '') { _%>
 
     private String <%= relationshipFieldName %><%= otherEntityFieldCapitalized %>;
@@ -201,11 +202,11 @@ public class <%= asDto(entityClass) %> implements Serializable {
     <%_ } else if (relationshipType === 'many-to-one' || (relationshipType === 'one-to-one' && ownerSide === true)) { _%>
 
     <%_ if (relationshipNameCapitalized.length > 1) { _%>
-    public <% if (relationshipFieldName === 'user' && authenticationType === 'oauth2') { %>String<% } else { %><%= primaryKeyType %><% } %> get<%= relationshipNameCapitalized %>Id() {
+    public <% if (otherEntityName === 'user' && authenticationType === 'oauth2') { %>String<% } else { %><%= primaryKeyType %><% } %> get<%= relationshipNameCapitalized %>Id() {
         return <%= relationshipFieldName %>Id;
     }
 
-    public void set<%= relationshipNameCapitalized %>Id(<% if (relationshipFieldName === 'user' && authenticationType === 'oauth2') { %>String<% } else { %><%= primaryKeyType %><% } %> <%= otherEntityName %>Id) {
+    public void set<%= relationshipNameCapitalized %>Id(<% if (otherEntityName === 'user' && authenticationType === 'oauth2') { %>String<% } else { %><%= primaryKeyType %><% } %> <%= otherEntityName %>Id) {
         this.<%= relationshipFieldName %>Id = <%= otherEntityName %>Id;
     }
     <%_ } else { // special case when the entity name has one character _%>
@@ -263,10 +264,11 @@ public class <%= asDto(entityClass) %> implements Serializable {
                 const relationshipFieldName = relationships[idx].relationshipFieldName;
                 const relationshipType = relationships[idx].relationshipType;
                 const relationshipNameCapitalized = relationships[idx].relationshipNameCapitalized;
+                const otherEntityName = relationships[idx].otherEntityName;
                 const otherEntityFieldCapitalized = relationships[idx].otherEntityFieldCapitalized;
                 const ownerSide = relationships[idx].ownerSide; _%>
                 <%_ if (relationshipType === 'many-to-one' || (relationshipType === 'one-to-one' && ownerSide === true)) { _%>
-            ", <%= relationshipFieldName %>Id=<% if (relationshipFieldName === 'user' && authenticationType === 'oauth2') { %>'<% } %>" + get<%= relationshipNameCapitalized %>Id() <% if (relationshipFieldName === 'user' && authenticationType === 'oauth2') { %>+ "'" <% } %>+
+            ", <%= relationshipFieldName %>Id=<% if (otherEntityName === 'user' && authenticationType === 'oauth2') { %>'<% } %>" + get<%= relationshipNameCapitalized %>Id() <% if (otherEntityName === 'user' && authenticationType === 'oauth2') { %>+ "'" <% } %>+
                     <%_ if (otherEntityFieldCapitalized !== 'Id' && otherEntityFieldCapitalized !== '') { _%>
             ", <%= relationshipFieldName %><%= otherEntityFieldCapitalized %>='" + get<%= relationshipNameCapitalized %><%= otherEntityFieldCapitalized %>() + "'" +
             <%_ } } } _%>

--- a/generators/entity-server/templates/src/main/resources/config/liquibase/changelog/added_entity.xml.ejs
+++ b/generators/entity-server/templates/src/main/resources/config/liquibase/changelog/added_entity.xml.ejs
@@ -160,12 +160,13 @@
             relationshipName = relationships[idx].relationshipName,
             ownerSide = relationships[idx].ownerSide,
             otherEntityName = relationships[idx].otherEntityName;
+            const relationshipColumnType = otherEntityName === 'user' && authenticationType === 'oauth2' ? 'varchar(100)' : 'bigint';
             if (relationshipType === 'many-to-many' && ownerSide) {
                 const joinTableName = getJoinTableName(entityTableName, relationshipName, prodDatabaseType);
           _%>
 
         <createTable tableName="<%= joinTableName %>">
-            <column name="<%= getColumnName(relationshipName) %>_id" type="bigint">
+            <column name="<%= getColumnName(relationshipName) %>_id" type="<%= relationshipColumnType %>">
                 <constraints nullable="false"/>
             </column>
             <column name="<%= getColumnName(name) %>_id" type="bigint">

--- a/generators/entity-server/templates/src/test/java/package/service/dto/EntityDTOTest.java.ejs
+++ b/generators/entity-server/templates/src/test/java/package/service/dto/EntityDTOTest.java.ejs
@@ -22,8 +22,9 @@ import org.junit.jupiter.api.Test;
 import static org.assertj.core.api.Assertions.assertThat;
 import <%= packageName %>.web.rest.TestUtil;
 <%_
-let id1 = getPkValueBasedOnDBAndAssociation(authenticationType, databaseType, relationships, '1');
-let id2 = getPkValueBasedOnDBAndAssociation(authenticationType, databaseType, relationships, '2');
+let primaryKeyType = getPkTypeBasedOnDBAndAssociation(authenticationType, databaseType, relationships);
+let id1 = getPrimaryKeyValue(primaryKeyType, databaseType, '1');
+let id2 = getPrimaryKeyValue(primaryKeyType, databaseType, '2');
 _%>
 <%_ if (id1.includes('UUID')) { _%>
 import java.util.UUID;

--- a/generators/entity-server/templates/src/test/java/package/service/dto/EntityDTOTest.java.ejs
+++ b/generators/entity-server/templates/src/test/java/package/service/dto/EntityDTOTest.java.ejs
@@ -22,12 +22,10 @@ import org.junit.jupiter.api.Test;
 import static org.assertj.core.api.Assertions.assertThat;
 import <%= packageName %>.web.rest.TestUtil;
 <%_
-let hasOauthUser = false;
-for (idx in relationships) {
-    hasOauthUser = relationships[idx].otherEntityName === 'user' && authenticationType === 'oauth2';
-}
+let id1 = getPkValueBasedOnDBAndAssociation(authenticationType, databaseType, relationships, '1');
+let id2 = getPkValueBasedOnDBAndAssociation(authenticationType, databaseType, relationships, '2');
 _%>
-<%_ if (databaseType === 'cassandra' || (databaseType === 'sql' && hasOauthUser === true)) { _%>
+<%_ if (id1.includes('UUID')) { _%>
 import java.util.UUID;
 <%_ } _%>
 
@@ -37,12 +35,12 @@ public class <%= asDto(entityClass) %>Test {
     public void dtoEqualsVerifier() throws Exception {
         TestUtil.equalsVerifier(<%= asDto(entityClass) %>.class);
         <%= asDto(entityClass) %> <%= asDto(entityInstance) %>1 = new <%= asDto(entityClass) %>();
-        <%= asDto(entityInstance) %>1.setId(<% if (databaseType === 'sql' && hasOauthUser === true) { %>UUID.randomUUID().toString()<% } else  if (databaseType === 'sql') { %>1L<% } else if (databaseType === 'mongodb' || databaseType === 'couchbase') { %>"id1"<% } else if (databaseType === 'cassandra') { %>UUID.randomUUID()<% } %>);
+        <%= asDto(entityInstance) %>1.setId(<%- id1 %>);
         <%= asDto(entityClass) %> <%= asDto(entityInstance) %>2 = new <%= asDto(entityClass) %>();
         assertThat(<%= asDto(entityInstance) %>1).isNotEqualTo(<%= asDto(entityInstance) %>2);
         <%= asDto(entityInstance) %>2.setId(<%= asDto(entityInstance) %>1.getId());
         assertThat(<%= asDto(entityInstance) %>1).isEqualTo(<%= asDto(entityInstance) %>2);
-        <%= asDto(entityInstance) %>2.setId(<% if (databaseType === 'sql' && hasOauthUser === true) { %>UUID.randomUUID().toString()<% } else  if (databaseType === 'sql') { %>2L<% } else if (databaseType === 'mongodb' || databaseType === 'couchbase') { %>"id2"<% } else if (databaseType === 'cassandra') { %>UUID.randomUUID()<% } %>);
+        <%= asDto(entityInstance) %>2.setId(<%- id2 %>);
         assertThat(<%= asDto(entityInstance) %>1).isNotEqualTo(<%= asDto(entityInstance) %>2);
         <%= asDto(entityInstance) %>1.setId(null);
         assertThat(<%= asDto(entityInstance) %>1).isNotEqualTo(<%= asDto(entityInstance) %>2);

--- a/generators/entity-server/templates/src/test/java/package/service/mapper/EntityMapperTest.java.ejs
+++ b/generators/entity-server/templates/src/test/java/package/service/mapper/EntityMapperTest.java.ejs
@@ -21,17 +21,11 @@ package <%= packageName %>.service.mapper;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import static org.assertj.core.api.Assertions.assertThat;
-<%
-let primaryKeyType = pkType;
-let hasOauthUser = false;
-for (idx in relationships) {
-    const otherEntityName = relationships[idx].otherEntityName;
-    const isUsingMapsId = relationships[idx].useJPADerivedIdentifier;
-    hasOauthUser = relationships[idx].otherEntityName === 'user' && authenticationType === 'oauth2';
-    primaryKeyType = (isUsingMapsId === true && otherEntityName === 'user' && authenticationType === 'oauth2') ? 'String' : primaryKeyType;
-}
-%>
-<%_ if (databaseType === 'sql' && hasOauthUser === true) { _%>
+<%_
+let primaryKeyType = getPkTypeBasedOnDBAndAssociation(authenticationType, databaseType, relationships);
+let id = getPkValueBasedOnDBAndAssociation(authenticationType, databaseType, relationships, '1');
+_%>
+<%_ if (id.includes('UUID')) { _%>
 import java.util.UUID;
 <%_ } _%>
 
@@ -46,7 +40,7 @@ public class <%= entityClass %>MapperTest {
 
     @Test
     public void testEntityFromId() {
-        <%= primaryKeyType %> id = <% if (databaseType === 'sql' && hasOauthUser === true) { %>UUID.randomUUID().toString()<% } else if (databaseType === 'sql') { %>2L<% } else if (databaseType === 'mongodb' || databaseType === 'couchbase') { %>"id2"<% } %>;
+        <%= primaryKeyType %> id = <%- id %>;
         assertThat(<%= entityInstance %>Mapper.fromId(id).getId()).isEqualTo(id);
         assertThat(<%= entityInstance %>Mapper.fromId(null)).isNull();
     }

--- a/generators/entity-server/templates/src/test/java/package/service/mapper/EntityMapperTest.java.ejs
+++ b/generators/entity-server/templates/src/test/java/package/service/mapper/EntityMapperTest.java.ejs
@@ -23,7 +23,7 @@ import org.junit.jupiter.api.Test;
 import static org.assertj.core.api.Assertions.assertThat;
 <%_
 let primaryKeyType = getPkTypeBasedOnDBAndAssociation(authenticationType, databaseType, relationships);
-let id = getPkValueBasedOnDBAndAssociation(authenticationType, databaseType, relationships, '1');
+let id = getPrimaryKeyValue(primaryKeyType, databaseType, '1');
 _%>
 <%_ if (id.includes('UUID')) { _%>
 import java.util.UUID;

--- a/generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs
+++ b/generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs
@@ -977,7 +977,7 @@ public class <%= entityClass %>ResourceIT <% if (databaseType === 'cassandra') {
 <%_ } _%>
         <%_ } _%>
         <%= entityInstance %>Repository.saveAndFlush(<%= asEntity(entityInstance) %>);
-        <% if (authenticationType === 'oauth2' && relationship.relationshipFieldName === 'user') { _%>String<%_ } else { _%>Long<% } %> <%= relationship.relationshipFieldName %>Id = <%= relationship.relationshipFieldName %>.getId();
+        <% if (authenticationType === 'oauth2' && relationship.otherEntityName === 'user') { _%>String<%_ } else { _%>Long<% } %> <%= relationship.relationshipFieldName %>Id = <%= relationship.relationshipFieldName %>.getId();
 
         // Get all the <%= entityInstance %>List where <%= relationship.relationshipFieldName %> equals to <%= relationship.relationshipFieldName %>Id
         default<%= entityClass %>ShouldBeFound("<%= relationship.relationshipFieldName %>Id.equals=" + <%= relationship.relationshipFieldName %>Id);

--- a/generators/generator-base-private.js
+++ b/generators/generator-base-private.js
@@ -1464,6 +1464,32 @@ module.exports = class extends Generator {
     }
 
     /**
+     * Returns the primary key value based on authentication type, DB, given association and default value
+     *
+     * @param {string} authenticationType - the auth type
+     * @param {string} databaseType - the database type
+     * @param {any} relationships - relationships
+     * @param {string} defaultValue - default value
+     */
+    getPkValueBasedOnDBAndAssociation(authenticationType, databaseType, relationships, defaultValue) {
+        const primaryKeyType = this.getPkTypeBasedOnDBAndAssociation(authenticationType, databaseType, relationships);
+        let value = defaultValue + 'L';
+        switch (primaryKeyType) {
+            case 'String':
+                value = '"id' + defaultValue + '"';
+                // Special case with a OneToOne relationship with User and @MapsId when using OAuth
+                if (databaseType === 'sql') {
+                    value = 'UUID.randomUUID().toString()';
+                }
+                break;
+            case 'UUID':
+                value = 'UUID.randomUUID()';
+                break;
+        }
+        return value;
+    }
+
+    /**
      * Get a root folder name for entity
      * @param {string} clientRootFolder
      * @param {string} entityFileName

--- a/generators/generator-base-private.js
+++ b/generators/generator-base-private.js
@@ -1473,10 +1473,10 @@ module.exports = class extends Generator {
      */
     getPkValueBasedOnDBAndAssociation(authenticationType, databaseType, relationships, defaultValue) {
         const primaryKeyType = this.getPkTypeBasedOnDBAndAssociation(authenticationType, databaseType, relationships);
-        let value = defaultValue + 'L';
+        let value;
         switch (primaryKeyType) {
             case 'String':
-                value = '"id' + defaultValue + '"';
+                value = `"id${defaultValue}"`;
                 // Special case with a OneToOne relationship with User and @MapsId when using OAuth
                 if (databaseType === 'sql') {
                     value = 'UUID.randomUUID().toString()';
@@ -1485,6 +1485,8 @@ module.exports = class extends Generator {
             case 'UUID':
                 value = 'UUID.randomUUID()';
                 break;
+            default:
+                value = `${defaultValue}L`;
         }
         return value;
     }

--- a/generators/generator-base-private.js
+++ b/generators/generator-base-private.js
@@ -1464,15 +1464,14 @@ module.exports = class extends Generator {
     }
 
     /**
-     * Returns the primary key value based on authentication type, DB, given association and default value
+     * Returns the primary key value based on the primary key type, DB and default value
      *
-     * @param {string} authenticationType - the auth type
+     * @param {string} primaryKeyType - the primary key type
      * @param {string} databaseType - the database type
-     * @param {any} relationships - relationships
      * @param {string} defaultValue - default value
+     * @returns {string} java primary key value
      */
-    getPkValueBasedOnDBAndAssociation(authenticationType, databaseType, relationships, defaultValue) {
-        const primaryKeyType = this.getPkTypeBasedOnDBAndAssociation(authenticationType, databaseType, relationships);
+    getPrimaryKeyValue(primaryKeyType, databaseType, defaultValue) {
         let value;
         switch (primaryKeyType) {
             case 'String':

--- a/generators/generator-base.js
+++ b/generators/generator-base.js
@@ -1375,6 +1375,12 @@ module.exports = class extends PrivateBase {
             );
 
             limit = 63;
+        } else if (prodDatabaseType === 'mariadb' && joinTableName.length > 64 && !this.skipCheckLengthOfIdentifier) {
+            this.warning(
+                `The generated join table "${joinTableName}" is too long for MariaDB (which has a 64 characters limit). It will be truncated!`
+            );
+
+            limit = 64;
         }
         if (limit > 0) {
             const halfLimit = Math.floor(limit / 2);

--- a/generators/generator-base.js
+++ b/generators/generator-base.js
@@ -1359,25 +1359,25 @@ module.exports = class extends PrivateBase {
         let limit = 0;
         if (prodDatabaseType === 'oracle' && joinTableName.length > 30 && !this.skipCheckLengthOfIdentifier) {
             this.warning(
-                `The generated join table "${joinTableName}" is too long for Oracle (which has a 30 characters limit). It will be truncated!`
+                `The generated join table "${joinTableName}" is too long for Oracle (which has a 30 character limit). It will be truncated!`
             );
 
             limit = 30;
         } else if (prodDatabaseType === 'mysql' && joinTableName.length > 64 && !this.skipCheckLengthOfIdentifier) {
             this.warning(
-                `The generated join table "${joinTableName}" is too long for MySQL (which has a 64 characters limit). It will be truncated!`
+                `The generated join table "${joinTableName}" is too long for MySQL (which has a 64 character limit). It will be truncated!`
             );
 
             limit = 64;
         } else if (prodDatabaseType === 'postgresql' && joinTableName.length >= 63 && !this.skipCheckLengthOfIdentifier) {
             this.warning(
-                `The generated join table "${joinTableName}" is too long for PostgreSQL (which has a 63 characters limit). It will be truncated!`
+                `The generated join table "${joinTableName}" is too long for PostgreSQL (which has a 63 character limit). It will be truncated!`
             );
 
             limit = 63;
         } else if (prodDatabaseType === 'mariadb' && joinTableName.length > 64 && !this.skipCheckLengthOfIdentifier) {
             this.warning(
-                `The generated join table "${joinTableName}" is too long for MariaDB (which has a 64 characters limit). It will be truncated!`
+                `The generated join table "${joinTableName}" is too long for MariaDB (which has a 64 character limit). It will be truncated!`
             );
 
             limit = 64;
@@ -1410,25 +1410,25 @@ module.exports = class extends PrivateBase {
         let limit = 0;
         if (prodDatabaseType === 'oracle' && constraintName.length >= 27 && !this.skipCheckLengthOfIdentifier) {
             this.warning(
-                `The generated constraint name "${constraintName}" is too long for Oracle (which has a 30 characters limit). It will be truncated!`
+                `The generated constraint name "${constraintName}" is too long for Oracle (which has a 30 character limit). It will be truncated!`
             );
 
             limit = 28;
         } else if (prodDatabaseType === 'mysql' && constraintName.length >= 61 && !this.skipCheckLengthOfIdentifier) {
             this.warning(
-                `The generated constraint name "${constraintName}" is too long for MySQL (which has a 64 characters limit). It will be truncated!`
+                `The generated constraint name "${constraintName}" is too long for MySQL (which has a 64 character limit). It will be truncated!`
             );
 
             limit = 62;
         } else if (prodDatabaseType === 'postgresql' && constraintName.length >= 60 && !this.skipCheckLengthOfIdentifier) {
             this.warning(
-                `The generated constraint name "${constraintName}" is too long for PostgreSQL (which has a 63 characters limit). It will be truncated!`
+                `The generated constraint name "${constraintName}" is too long for PostgreSQL (which has a 63 character limit). It will be truncated!`
             );
 
             limit = 61;
         } else if (prodDatabaseType === 'mariadb' && constraintName.length >= 61 && !this.skipCheckLengthOfIdentifier) {
             this.warning(
-                `The generated constraint name "${constraintName}" is too long for MariaDB (which has a 64 characters limit). It will be truncated!`
+                `The generated constraint name "${constraintName}" is too long for MariaDB (which has a 64 character limit). It will be truncated!`
             );
 
             limit = 62;

--- a/generators/generator-base.js
+++ b/generators/generator-base.js
@@ -1426,6 +1426,12 @@ module.exports = class extends PrivateBase {
             );
 
             limit = 61;
+        } else if (prodDatabaseType === 'mariadb' && constraintName.length >= 61 && !this.skipCheckLengthOfIdentifier) {
+            this.warning(
+                `The generated constraint name "${constraintName}" is too long for MariaDB (which has a 64 characters limit). It will be truncated!`
+            );
+
+            limit = 62;
         }
         if (limit > 0) {
             const halfLimit = Math.floor(limit / 2);

